src/OrchardCore.Modules/OrchardCore.Setup/Views/_Layout.cshtml

```diff
@@ -21,5 +57,22 @@
             @await RenderSectionAsync("Footer", required: false)
         </footer>
     </div>
+    <script>
+        function setLocalizationCookie() {
+            var culturesList = document.getElementById('culturesList');
+            var culture = culturesList.options[culturesList.selectedIndex].value
+            var url = window.location.origin;
+
+            if (url.indexOf('?') > -1) {
+                url = url.split('&ui-culture=')[0] + '&';
+            }
+            else {
+                url += '?';
+            }
+
+            url += 'ui-culture=' + culture;
+            window.location = url;
```

Comment:
i don't remember all details and maybe not so important, so just for infos and as i remember.

Generating / updating url is not so simple, the url may have other parameters (not only `ui-culture`) the url may end with a fragment string `#...` and so on. And i don't talk about encoding because i assume that the culture keys / values don't need to be encoded.

Take a look e.g at what the `QueryHelpers.AddQueryString()` [does here](https://github.com/dotnet/aspnetcore/blob/116799fa709ff003781368b578e4efe2fa32e937/src/Http/WebUtilities/src/QueryHelpers.cs#L85).

Reply:
FYI I didn't ignore the currents params if there is

Reply:
I agree, but as of now, we don't have any other params on this url ; so I don't think it will be an issue.

Reply:
> FYI I didn't ignore the currents params if there are

Again maybe not so important but the part of your code that checks an existing `ui-culture` query parameter and then replaces it, looks wrong for me.

Parsing / updating a query string is not so simple.

E.g you are splitting on `&ui-culture=` so seems that you only keep the previous parameter(s), and what happens if `ui-culture=` is the 1st parameter `...?ui-culture=...` without starting with the `&` separator.

Reply:
> what happens if ui-culture= is the 1st parameter ...?ui-culture=... without starting with the & separator.

Probably I forgot this case

> I agree, but as of now, we don't have any other params on this url ; so I don't think it will be an issue.

There will be a params if you 're use the setup from tenants page

Reply:
Maybe the [`URLSearchParams` interface might be helpful](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)

Reply:
Ok you guys fix that part. I added the required changes requested by @sebastienros

Reply:
Noo, that's why my push refuse ;)